src/modules/geometry.cxx

```diff
@@ -1,30 +1,46 @@
 module;
 
+#include <vector>
+#include <cstdint>
+
+template<typename T>
+concept Number = std::integral<T> || std::floating_point<T>;
 
 export module Geometry;
 
-export struct Vector_2d
+export template<Number Component_type, std::unsigned_integral auto _dimensions>
+struct Vector
 {
-    double x, y;
-    Vector_2d () = default;
-    Vector_2d(double x, double y) : x(x), y(y) {}
-
+    std::vector<Component_type> components{};
+    constexpr static auto dimensions = _dimensions;
 };
 
-export typedef Vector_2d Point;
+export template<Number Component_type>
+using Vector_2d = Vector<Component_type, (uint8_t)2>;
```

Comment:
I find the cast here a bit annoying, I'm sure the users of this vector won't appreciate it. Maybe make dimensions an int template parameter.